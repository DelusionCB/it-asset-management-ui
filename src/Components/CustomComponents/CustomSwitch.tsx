import React from 'react';
import {Col, FormGroup, Input, Label} from 'reactstrap';
import {useTranslation} from 'react-i18next';
import './index.scss'
import {CustomSwitchProps} from './types.customcomps';

function CSwitch ({value, label, validation, onChange, id, disabled, invalid}: CustomSwitchProps): JSX.Element {
    const {t} = useTranslation()

    return (
        <Col className='switches'>
            <FormGroup switch>
                <Label check for={id}>
                    {t(label)}
                </Label>
                <Input
                    aria-checked={value}
                    id={id}
                    disabled={disabled}
                    invalid={invalid}
                    aria-invalid={invalid}
                    name={id}
                    type='switch'
                    checked={value}
                    onChange={(e) => { onChange(e); }}
                />
            </FormGroup>
        </Col>
    );
}

export default CSwitch;
