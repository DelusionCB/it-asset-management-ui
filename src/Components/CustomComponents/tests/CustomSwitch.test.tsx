import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import CustomSwitch from '../CustomSwitch';
import '@testing-library/jest-dom'

describe('<CustomSwitch />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <CustomSwitch disabled={false} id={'id'} label={'label'} onChange={jest.fn()} value={false}/>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
