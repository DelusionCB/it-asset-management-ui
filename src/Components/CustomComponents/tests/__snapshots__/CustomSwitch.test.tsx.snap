// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomSwitch /> Snapshot renders correctly 1`] = `
<div
  className="switches col"
>
  <div
    className="form-check form-switch"
  >
    <label
      className="form-check-label"
      htmlFor="id"
    >
      label
    </label>
    <input
      aria-checked={false}
      aria-invalid={false}
      checked={false}
      className="form-check-input"
      disabled={false}
      id="id"
      name="id"
      onChange={[Function]}
      type="checkbox"
    />
  </div>
</div>
`;
